source/api_cc/include/common.h

```diff
@@ -32,9 +32,13 @@ typedef double ENERGYTYPE;
 
 struct NeighborListData 
 {
+  /// Number of core region atoms
   std::vector<int > ilist;
+  /// Array stores the core region atom's index
   std::vector<std::vector<int> > jlist;
+  /// Array stores the core region atom's neighbor atom number
   std::vector<int > numneigh;
+  /// Array stores the core region atom's neighbor index
```

Comment:
this comment is for `numneigh`. Place it above `std::vector<int > numneigh;`? 

The dsame issue for `ilist` and `jlist`

Reply:
In #968, I see @denghuilu add the same comment to `InputNlist::firstneigh`? Isn't it the same thing as `NeighborListData::firstneigh`? Did I understand wrongly?

Reply:
@denghuilu @njzjz  `jlist` stores the indexes of all the neighbors. `firstneigh[i]` stores the location of the first neighbor of `i`. So the neighbors of `i` are stored in `jlist[firstneigh[i] : firstneigh[i] + numneigh[i]`

@denghuilu Please check again you code you really use the neighbor list correctly....

Reply:
@amcadmus I think you are right, so the NeighborListData is used to construct the InputNlist. From my point of view, firstneigh is a two-dimensional array that stores neighbor atom's index, even if it only refers to a series of jlist pointers. I'm using the InputNlist in the custom op, so I think it's fine.